Java/core/src/main/java/com/amazon/randomcutforest/tree/AbstractRandomCutTree.java

```diff
@@ -270,75 +270,99 @@ void updateAncestorNodesAfterDelete(NodeReference nodeReference, Point point) {
         updateAncestorPointSum(nodeReference);
     }
 
+    NodeReference findLeaf(Point point, boolean deleteMass) {
+        NodeReference nodeReference = root;
+        while (!isLeaf(nodeReference)) {
+            if (deleteMass) {
+                decrementMass(nodeReference);
+            }
+            nodeReference = (leftOf(point, getCutDimension(nodeReference), getCutValue(nodeReference)))
+                    ? getLeftChild(nodeReference)
+                    : getRightChild(nodeReference);
+        }
+        return nodeReference;
+    }
+
+    NodeReference findLeafAndVerify(Point point, boolean deleteMass) {
+        NodeReference nodeReference = findLeaf(point, deleteMass);
+        Point oldPoint = getPointFromLeafNode(nodeReference);
+        if (!equals(oldPoint, point)) {
+            throw new IllegalStateException(toString(point) + " " + toString(getPointFromLeafNode(nodeReference)) + " "
+                    + nodeReference + " node " + false + " Inconsistency in trees in delete step here.");
+        }
+        return nodeReference;
+    }
+
     /**
-     * method to delete a point from the tree
+     * the following function returns the number of copies of a point in the tree
+     * and switches the reference to the provided reference. This may be useful for
+     * collating duplicate points across trees. By default this is set to no-op.
      *
+     * @param leafReference reference of the leaf node
+     * @param newRef        reference of the point stored at the leaf node
+     * @return the number of copies of the point at leaf node
      */
+    abstract void switchLeafReference(NodeReference leafReference, PointReference newRef);
 
-    @Override
-    public void deletePoint(PointReference pointReference, long sequenceNumber) {
-        checkState(root != null, "root must not be null");
-        deletePoint(root, getPointFromPointReference(pointReference), sequenceNumber, 0);
+    public void switchLeafReference(PointReference newRef) {
+        checkNotNull(newRef, " cannot be null ");
+        NodeReference nodeReference = findLeafAndVerify(getPointFromPointReference(newRef), false);
+        switchLeafReference(nodeReference, newRef);
+    }
+
+    public int getCopies(PointReference reference, boolean flagError) {
```

Comment:
This method could use documentation explaining what kind of error is being checked for with `flagError`.

Reply:
Changed the method to avoid error flags etc.